uploader/pantheon.py

```diff
@@ -161,6 +157,146 @@ def remove_trailing_slash(path):
         path = path[:-1]
     return path
 
+def recursive_glob(directory, pattern='*'):
+    if not os.path.exists(directory):
+        raise ValueError("Directory not found {}".format(directory))
+
+    matches = []
+    for root, dirnames, filenames in os.walk(directory):
+        for filename in filenames:
+            if filename == 'pantheon2.yml':
+                continue
+            full_path = os.path.join(root, filename)
+            if fnmatch.filter(full_path, pattern):
+                matches.append(os.path.join(root, filename))
+    return matches
+
+def define_resource(path, titleGlobs, moduleGlobs, resourceGlobs):
+    isTitle = matches(path, titleGlobs, 'titles')
+    isModule = matches(path, moduleGlobs, 'modules') if not isTitle else False
+    isResource = matches(path, resourceGlobs, 'resources') if not isModule else False
+
+    if isTitle:
+        resouce_type = "titles"
+    if isModule:
+        resouce_type = "modules"
+    if isResource:
+        resouce_type = "resources"
+    return resource_type
+
+def is_slash_wildcard(pattern):
+    return bool(re.match(r"^.*([a-z]?[/]+[*]?$)", pattern.lower()))
+
+def get_subdir(pattern):
+    delimiter = '/*'
+    pos = pattern.find(delimiter)
+    return pattern[0:pos]
+
+def find_files(patterns, directory):
+    files = []
+
+    if patterns:
+        for pattern in patterns:
+            if is_slash_wildcard(pattern):
+                subdir = get_subdir(pattern)
+                #Use recursive_glob(directory, pattern) for /*
+                items = recursive_glob(directory + '/' + subdir)
+                print("pattern in globs: ", pattern)
+                print("items found: ", items)
+                files = files + items
+            else:
+                items = []
+                for file in glob.iglob(directory + '/**/' + pattern, recursive=True):
+                    items.append(file)
+                if len(items) > 0:
+                    files = files + items
+    return files
+
+def process_file(path, type):
+    global processed_files
+    isTitle = True if type == 'titles' else False
+    isModule = True if type == 'modules' else False
+    isResource = True if type =='resources' else False
+    url = server + "/content/repositories/" + repository
+
+    if isModule or isTitle or isResource:
+        path = PurePath(path)
+        base_name = path.stem
+
+        ppath = path
+        hiddenFolder = False
+        while not ppath == PurePath(args.directory):
+            logger.debug('ppath: %s', str(ppath.stem))
+            if ppath.stem[0] == '.':
+                hiddenFolder = True
+                break
+            ppath = ppath.parent
+        if hiddenFolder:
+            logger.info('Skipping %s because it is hidden.', str(path))
+            logger.info('')
```

Comment:
I would make this a debug level statement.... for a quieter console.

Reply:
updated.